src/Sylius/Behat/Context/Setup/ProductContext.php

```diff
@@ -170,6 +206,127 @@ public function productVariantBelongsToTaxCategory(
+
+        $this->objectManager->flush();
+    }
+
+    /**
+     * @Given /^(this product) has ([^"]+) attribute "([^"]+)" set to "([^"]+)"$/
+     */
+    public function thisProductHasCheckboxAttributeWithValue(ProductInterface $product, $productAttributeType, $productAttributeName, $value)
+    {
+        $this->createProductAttribute($productAttributeType, $productAttributeName);
+        $booleanValue = ('Yes' === $value);
+        $attributeValue = $this->createProductAttributeValue($booleanValue);
+        $product->addAttribute($attributeValue);
+
+        $this->objectManager->flush();
+    }
+
+    /**
+     * @Given /^(this product) has ([^"]+) attribute "([^"]+)" with date "([^"]+)"$/
+     */
+    public function thisProductHasDateTimeAttributeWithDate(ProductInterface $product, $productAttributeType, $productAttributeName, $date)
+    {
+        $this->createProductAttribute($productAttributeType, $productAttributeName);
+        $attributeValue = $this->createProductAttributeValue(new \DateTime($date));
+
+        $product->addAttribute($attributeValue);
+
+        $this->objectManager->flush();
+    }
+
+    /**
+     * @Given /^(this product) has option "([^"]*)" with value "([^"]*)" priced at ("[^"]+")$/
+     */
+    public function thisProductHasOption(ProductInterface $product, $optionName, $value, $price)
+    {
+        /** @var OptionInterface $variant */
+        $option = $this->productOptionFactory->createNew();
+
+        $option->setName($optionName);
+        $option->setCode('PO1');
+
+        /** @var OptionValueInterface $optionValue */
+        $optionValue = $this->productOptionValueFactory->createNew();
+
+        $optionValue->setValue($value);
+        $optionValue->setCode('POV1');
+        $optionValue->setOption($option);
+
+        $option->addValue($optionValue);
+
+        /** @var ProductVariantInterface $variant */
+        $variant = $this->productVariantFactory->createNew();
+
+        $variant->setPrice($price);
+        $variant->addOption($optionValue);
+        $variant->setProduct($product);
+
+        $product->addOption($option);
+        $product->addVariant($variant);
+        $product->setVariantSelectionMethod(ProductInterface::VARIANT_SELECTION_MATCH);
+
+        $this->sharedStorage->set(sprintf('%s_option',$optionName), $option);
+
+        $this->objectManager->persist($option);
+        $this->objectManager->persist($optionValue);
+        $this->objectManager->flush();
+    }
+
+
+    /**
+     * @param string $type
+     * @param string $name
+     * @param string $code
+     */
+    private function createProductAttribute($type, $name, $code = 'PA112')
+    {
+        $productAttribute = $this->productAttributeFactory->createTyped($type);
+        $productAttribute->setCode($code);
+        $productAttribute->setName($name);
+
+        $this->objectManager->persist($productAttribute);
+        $this->sharedStorage->set('product_attribute', $productAttribute);
```

Comment:
Why not returning the instance here? Using shared storage here sounds like an overkill.